src/eh1/pin.rs

```diff
@@ -249,4 +328,30 @@ mod test {
 
         pin.done();
     }
+
+    #[test]
+    fn test_stateful_output_pin() {
+        let expectations = [
+            Transaction::new(GetSetState(State::Low)),
+            Transaction::new(Toggle),
+            Transaction::get_set_state(State::High),
+            Transaction::toggle(),
+            Transaction::new(GetSetState(State::Low))
+                .with_error(MockError::Io(ErrorKind::NotConnected)),
+            Transaction::new(Toggle).with_error(MockError::Io(ErrorKind::NotConnected)),
```

Comment:
Why not use the factory methods instead of passing in the transaction kinds directly?

Reply:
I was just testing both ways for completion and this is the way that the other tests define transactions so I followed suit.

Reply:
Thanks! Fixup commits are fine as well, I can squash them before merging 🙂